content/exporters/supported-exporters/Java/ocagent.md

```diff
@@ -0,0 +1,440 @@
+- [References](#references)
+
+### Introduction
+The OpenCensus Agent exporter aka "ocagent-exporter" enables Java applications to send the observability
+that they've collected using OpenCensus to the [OpenCensus Agent](https://github.com/census-instrumentation/opencensus-service)
+
+This exporter connects and sends observability signals via a single HTTP/2 stream and gRPC with Protocol Buffers
+to the OpenCensus Agent. If unspecified, this exporter tries to connect to the OpenCensus Agent on port `55678`.
+
+### Purpose
+It converts OpenCensus Stats and Traces into OpenCensus Proto Metrics and Traces which are then sent to the OpenCensus Agent.
+Therefore programs no longer have to enable the traditional backends' exporters for every single application.
+
+For example, some backends like Prometheus require opening a unique port per application. This port is what they'll pull stats from.
+If you have 10,000 microservices that all each export to Prometheus, you already have to manually and uniquely create
+at least 10,000 unique ports. Uniquely creating a port will not only exhaust your file descriptors but it also becomes
+cumbersome and error prone to do.
+
+With the ocagent-exporter, instead the stats are uploaded to the OpenCensus Agent and from there,
+the agent is singly scraped by Prometheus.
+
+As you can see from above, this not only scales your application's resource allocation but also scales your development speed
+and liberates you from complex batching deployments. It also ensures that your applications can be kept light, that the agent's
+deployment can safely be restarted flexibly.
+
+The same thing happens for traces.
+
+### Add the dependencies to your project
+
+For Maven add to your pom.xml:
+
+{{<highlight xml>}}
+<dependencies>
+  <dependency>
+    <groupId>io.opencensus</groupId>
+    <artifactId>opencensus-exporter-metrics-ocagent</artifactId>
+    <version>0.20.0</version>
+  </dependency>
+	<dependency>
+    <groupId>io.opencensus</groupId>
+    <artifactId>opencensus-exporter-trace-ocagent</artifactId>
+    <version>0.20.0</version>
+  </dependency>
+</dependencies>
+{{</highlight>}}
+
+For Gradle add to your dependencies:
+
+{{<highlight groovy>}}
+compile 'io.opencensus:opencensus-exporter-metrics-ocagent:0.20.0'
+compile 'io.opencensus:opencensus-exporter-trace-ocagent:0.20.0'
+{{</highlight>}}
+
+### Register the exporter
+
+An exporter can be started by invoking `createAndRegister()`, whose signature is:
+{{<highlight java>}}
+OcAgentMetricsExporter.createAndRegister(OcAgentMetricsExporterConfiguration.builder().build());
+OcAgentTraceExporter.createAndRegister(OcAgentTraceExporterConfiguration.builder().build());
+{{</highlight>}}
+
+### Options
+Options allow you to customize the exporter.
+Options use a builder pattern which allows optional functional options.
+{{<highlight java>}}
+OcAgentTraceExporterConfiguration.builder().setEndPoint("localhost:8888").setUseInsecure(true).build();
```

Comment:
Let's not use `localhost:8888` but instead `localhost:55678`

Reply:
SG, done.